ease-lab/vhive
# gg port

To test using stock Knative:
1) Run the stock-only setup scripts (https://github.com/faromero/vhive/blob/main/docs/developers_guide.md#testing-stock-knative-setup-or-images)
2) Run "kn service apply gg-port-0 --filename configs/knative_workloads/gg_port.yaml"
Note: This might require for you to run sudo chown $(id -u):$(id -g) $HOME/.kube/config due to file access permissions if the config file can't be loaded
3) Run "curl -v -X POST url" using the url outputted by the above command (It should respond with a JSON error)

To test using gg input (in progress):
1) Clone vhive_port branch from http://github.com/faromero/gg
2) Run setup (https://github.com/StanfordSNR/gg#build-directions)
3) Enter a test directory, such as the one in tests/hello_world, or mosh directory and run gg init, and gg infer make
4) Run "gg force --jobs 1 --engine vhive=(IP or url):port executable" to compile the executable
Note: This is where I have been trying to get a working url or IP for the ingress gateway. I have been testing with curl commands as I tried to debug, so that would be another way to test for a working connection with the ingress gateway.
